### PR TITLE
Promote Event CRUD tests to conformance

### PR DESCRIPTION
**Pull request by chelseychen**
**What type of PR is this?**

/kind api-change

**What this PR does / why we need it**:

The new Event API has been graduated to v1, now its CRUD tests should also be promoted to conformance.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
- [KEP]: <https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/383-new-event-api-ga-graduation>
```

### CODE REVIEW

File: test/e2e/instrumentation/events.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ var _ = common.SIGDescribe("Events API", func() {
 		framework.ExpectEqual(foundCreatedEvent, false, "should not have found test event after deletion")
 	})
 
-	ginkgo.It("should delete a collection of events", func() {
+	framework.ConformanceIt("should delete a collection of events", func() {
```
Comment (wojtek-t) [inline test/e2e/instrumentation/events.go:146]: If you add it to conformance you need to add a similar comment as in the test above (descirbing what the test is doing).

File: test/e2e/instrumentation/events.go [hunk 2]
```diff
@@ -88,7 +88,7 @@ var _ = common.SIGDescribe("Events API", func() {
 		The event is patched with a new note, the check MUST have the update note.
 		The event is deleted and MUST NOT show up when listing all events.
 	*/
-	ginkgo.It("should ensure that an event can be fetched, patched, deleted, and listed", func() {
+	framework.ConformanceIt("should ensure that an event can be fetched, patched, deleted, and listed", func() {
```
Comment (wojtek-t) [inline test/e2e/instrumentation/events.go:91]: You also need to add this to test/conformance/testdata/conformance.yaml

File: test/e2e/instrumentation/events.go (hunk 1 above)
Comment (chelseychen) [reply test/e2e/instrumentation/events.go:146]: Done.

File: test/e2e/instrumentation/events.go (hunk 2 above)
Comment (chelseychen) [reply test/e2e/instrumentation/events.go:91]: Done.

**Comment by wojtek-t** [comment]
@chelseychen - the bazel-test failure seems like a valid one, please fix

**Comment by johnbelamaric** [comment]
How long have these tests existed? I only see one week of history in https://testgrid.k8s.io/google-gci#gce-cos-master-flaky-repro&width=5&include-filter-by-regex=sig-instrumentation

We need two weeks before merging.

Note that code freeze is Thursday, but Test Freeze is not until August 6th, so there is plenty of time to get this in.

https://github.com/kubernetes/sig-release/tree/master/releases/release-1.19#timeline

**Comment by chelseychen** [comment]
> How long have these tests existed? I only see one week of history in https://testgrid.k8s.io/google-gci#gce-cos-master-flaky-repro&width=5&include-filter-by-regex=sig-instrumentation

Right, the tests were just added a week ago: #92607 

> We need two weeks before merging.
> 
> Note that code freeze is Thursday, but Test Freeze is not until August 6th, so there is plenty of time to get this in.
> 
> https://github.com/kubernetes/sig-release/tree/master/releases/release-1.19#timeline

SGTM, thanks for your review.

**Comment by Riaankl** [comment]
assign @fabxc
Can you please review and /approve

**Comment by Riaankl** [comment]
assign @wojtek-t
Can you please review and /approve

**Comment by hh** [comment]
/retest
rinse / repeat in two hours... when the timeout hits

**Pull request merged**